lib/model-access-wrapper.js

```diff
@@ -749,18 +789,26 @@ class ModelAccessWrapper extends CrispHooks {
 		let prof = profiler.begin('#update');
 
 		let model = this.model;
-		let query, update;
 		return Promise.resolve()
 			.then(() => this._preprocessParams(params, 'update'))
 			.then(() => this.trigger('pre-update', params))
 			.then(() => {
-				query = createQuery(params.query, {
+				const query = createQuery(params.query, {
 					schema: model.getSchema && model.getSchema()
 				});
-				update = createUpdate(params.update, {
+				const update = createUpdate(params.update, {
 					allowFullReplace: false,
 					schema: model.getSchema && model.getSchema()
 				});
+				const updatedFields = update.getUpdatedFields();
+
+				if (this.hasProtectedFields && !params.overwriteProtected) {
+					const containsProtected = this.protectedFields.some((field) => _.includes(updatedFields, field));
```

Comment:
Could also break with arrays.

Also another way of doing this would be to use _.intersection but that might also break with arrays.